holidays/countries/thailand.py

```diff
@@ -95,54 +95,54 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
         * Coronation Day:
-            * <https://www.matichon.co.th/politics/news_526200>
-            * <https://workpointtoday.com/news1-5/>
-        * [HM Queen Suthida's Birthday](https://www.thairath.co.th/news/politic/1567418)
-        * [HM Maha Vajiralongkorn's Birthday](https://www.matichon.co.th/politics/news_526200)
-        * [HM Queen Sirikit the Queen Mother's Birthday](https://hilight.kapook.com/view/14164)
-        * [National Mother's Day](https://www.brh.go.th/index.php/2019-02-27-04-11-52/542-12-2564)
-        * [HM King Bhumibol Adulyadej Memorial Day](https://www.matichon.co.th/politics/news_526200)
+            * <https://web.archive.org/web/20220817223130/https://www.matichon.co.th/politics/news_526200>
+            * <https://web.archive.org/web/20230205075326/https://workpointtoday.com/news1-5/>
+        * [HM Queen Suthida's Birthday](https://web.archive.org/web/20241217104142/https://www.thairath.co.th/news/politic/1567418)
+        * [HM Maha Vajiralongkorn's Birthday](https://web.archive.org/web/20220817223130/https://www.matichon.co.th/politics/news_526200)
+        * [HM Queen Sirikit the Queen Mother's Birthday](https://web.archive.org/web/20250203002257/https://hilight.kapook.com/view/14164)
+        * [National Mother's Day](https://web.archive.org/web/20221207161434/https://www.brh.go.th/index.php/2019-02-27-04-11-52/542-12-2564)
+        * [HM King Bhumibol Adulyadej Memorial Day](https://web.archive.org/web/20220817223130/https://www.matichon.co.th/politics/news_526200)
         * HM King Chulalongkorn Memorial Day:
             * <https://th.wikipedia.org/wiki/วันปิยมหาราช>
-            * <https://www.sanook.com/news/9072518/>
+            * <https://web.archive.org/web/20231024220855/https://www.sanook.com/news/9072518/>
         * HM King Bhumibol Adulyadej's Birthday
             * [Ministry of Culture](http://tiny.cc/wa_mincul_nat_day)
-            * <https://hilight.kapook.com/view/148862>
-        * [National Father's Day](https://www.brh.go.th/index.php/2019-02-27-04-12-21/594-5-5)
+            * <https://web.archive.org/web/20250126073943/https://hilight.kapook.com/view/148862>
+        * [National Father's Day](https://web.archive.org/web/20230205075650/https://www.brh.go.th/index.php/2019-02-27-04-12-21/594-5-5)
         * Constitution Day:
             * [ประเทศไทย](https://th.wikipedia.org/wiki/วันรัฐธรรมนูญ_)
-            * <https://hilight.kapook.com/view/18208>
+            * <https://web.archive.org/web/20250212210436/https://hilight.kapook.com/view/18208>
             * [Bank of Thailand](http://tiny.cc/wa_bot_1992)
-            * <https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx>
+            * <https://web.archive.org/web/20161028001043/http://www.myhora.com:80/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99-%E0%B8%9E.%E0%B8%A8.2475.aspx>
         * New Year's Eve:
             * [Bank of Thailand](http://tiny.cc/wa_bot_1992)
```

Comment:
All tiny.cc sources are just shortened Internet Archive links, AFAIK I did used them for Thailand, Canada, and South Africa back when I first joined to prevent links from going over character width limit - will need to double check them later
```suggestion
            * <https://web.archive.org/web/20230205072056/https%3A//www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx>
```